lib/beacon/runtime_js.ex

```diff
@@ -58,17 +61,62 @@ defmodule Beacon.RuntimeJS do
 
     gzip = :zlib.gzip(js)
 
-    if :ets.insert(:beacon_assets, {:js, {hash, js, brotli, gzip}}) do
+    if :ets.insert(:beacon_assets, {{site, :js}, {hash, js, brotli, gzip}}) do
       :ok
     else
       raise Beacon.LoaderError, "failed to compress js"
     end
   end
 
-  def current_hash do
-    case :ets.match(:beacon_assets, {:js, {:"$1", :_, :_, :_}}) do
+  def current_hash(site) do
+    case :ets.match(:beacon_assets, {{site, :js}, {:"$1", :_, :_, :_}}) do
       [[hash]] -> hash
       _ -> nil
     end
   end
+
+  def build_hooks(site, minify?) do
+    joiner = if(minify?, do: ",", else: ",\n")
+
+    site
+    |> Beacon.Content.list_js_hooks()
+    |> Enum.map_join(joiner, fn hook ->
+      code =
+        IO.iodata_to_binary([
+          if(hook.mounted, do: format_callback(hook, :mounted, minify?), else: []),
+          if(hook.beforeUpdate, do: format_callback(hook, :beforeUpdate, minify?), else: []),
+          if(hook.updated, do: format_callback(hook, :updated, minify?), else: []),
+          if(hook.destroyed, do: format_callback(hook, :destroyed, minify?), else: []),
+          if(hook.disconnected, do: format_callback(hook, :disconnected, minify?), else: []),
+          if(hook.reconnected, do: format_callback(hook, :reconnected, minify?), else: [])
+        ])
+
+      if minify? do
+        "#{hook.name}:{#{code}}"
+      else
+        "    #{hook.name}: {\n#{code}    }"
+      end
+    end)
+  end
+
+  defp format_callback(hook, callback, minify?)
+
+  defp format_callback(hook, callback, false) do
+    IO.iodata_to_binary([
+      "      ",
+      to_string(callback),
+      "() {\n        ",
+      hook |> Map.fetch!(callback) |> String.trim_trailing() |> String.replace("\n", "\n        "),
+      "\n      },\n"
+    ])
+  end
+
+  defp format_callback(hook, callback, true) do
+    IO.iodata_to_binary([
+      to_string(callback),
+      "(){",
+      hook |> Map.fetch!(callback) |> String.replace("\n", ""),
+      "},"
+    ])
+  end
```

Comment:
You could simplify this code and remove unnecessary calls, something like:

```suggestion
      code =
        [
          format_callback(hook, :mounted, minify?),
          # and ther rest...
        ])

      code =
      if minify? do
        [hook.name, ":{", code, "}"]
      else
        ["    ", hook.name, ": {\n", code, "\n    }"]
      end
      
      IO.iodata_to_binary(code)
    end)
  end

  defp format_callback(hook, callback, minify?)

  defp format_callback(nil = _hook, _callback, _minify?), do: []
 
  defp format_callback(hook, callback, false) do
    [
      "      ",
      to_string(callback),
      "() {\n        ",
      hook |> Map.fetch!(callback) |> String.trim_trailing() |> String.replace("\n", "\n        "),
      "\n      },\n"
    ]
  end

  defp format_callback(hook, callback, true) do
    [
      to_string(callback),
      "(){",
      hook |> Map.fetch!(callback) |> String.replace("\n", ""),
      "},"
    ]
  end
```